internal/docstore/drivertest/drivertest.go

```diff
@@ -299,47 +317,117 @@ func testCodec(t *testing.T, ct CodecTester) {
 		t.Skip("no CodecTester")
 	}
 
-	type S struct {
+	check := func(in, dec interface{}, encode func(interface{}) (interface{}, error), decode func(interface{}, interface{}) error) {
+		t.Helper()
+		enc, err := encode(in)
+		if err != nil {
+			t.Fatal(err)
+		}
+		if err := decode(enc, dec); err != nil {
+			t.Fatal(err)
+		}
+		if diff := cmp.Diff(in, dec); diff != "" {
+			t.Error(diff)
+		}
+	}
+
+	type DocstoreRoundTrip struct {
 		N  *int
 		I  int
+		U  uint
 		F  float64
+		C  complex128
 		St string
 		B  bool
 		By []byte
 		L  []int
+		A  [2]int
 		M  map[string]bool
+		P  *string
+		T  time.Time
 	}
-	// TODO(jba): add more fields: more basic types; pointers; structs; embedding.
+	// TODO(jba): add more fields: structs; embedding.
 
-	in := S{
+	s := "bar"
```

Comment:
Consider adding a comment here, explaining that the docstore round trip should work for all docstore-supported types, regardless of native driver support.

Reply:
Done.